### PR TITLE
Job List Limit & Offset

### PR DESCRIPTION
**Pull request by JVickery-TBS**
feat(logic): limit and offset for job list;

- Added params to `job_list` logic action method.

@wardi seems like there is kind of an issue with how to limit/offset the job lists here, because of the case of multiple queue. It is fine for us of course because we only use on Redis queue. But when putting this upstream, I am not sure how to get around that :/

### CODE REVIEW

**Comment by JVickery-TBS** [comment]
as discussed, the issue is doing an offset/limit across queues.

Ideas are:

1. merge all the queue jobs so we can order by date created and change to time based limit/offset;
2. OR do not support limit/offset for multiple queues.

Preferably, we do not really want to get every single job and merge them when we just do something like `limit=20`. Investigate the RQ (redis-queue) py lib to see what are options are.

We are going for best performance here, while maintaining/improving any output/return for the `job_list` action method.

**Comment by JVickery-TBS** [comment]
@wardi okay this is good for review now. So `job_list` action method now supports limit and ids_only. The default limit is now 200. And the list returned will behave just like the RQ workers will do the order of execution.

File: ckan/logic/action/get.py [hunk 1]
```diff
@@ -3532,22 +3532,36 @@ def job_list(context, data_dict):
...
+    Thus the left most non-empty queue will be emptied firts
+    before the next right non-empty one.
+
     .. versionadded:: 2.7
     '''
     _check_access(u'job_list', context, data_dict)
-    dictized_jobs = []
+    jobs_list = []
     queues = data_dict.get(u'queues')
+    limit = data_dict.get(u'limit',
```
Comment (wardi) [inline ckan/logic/action/get.py:3551]: Probably better to validate the parameters and return proper validation errors, but here you at least need to make sure this is an int

File: ckan/logic/action/get.py [hunk 2]
```diff
@@ -3532,22 +3532,36 @@ def job_list(context, data_dict):
...
         queues = jobs.get_all_queues()
     for queue in queues:
-        for job in queue.jobs:
-            dictized_jobs.append(jobs.dictize_job(job))
-    return dictized_jobs
+        if ids_only:
+            jobs_list += queue.job_ids
+        else:
+            jobs_list += [jobs.dictize_job(j) for j in queue.jobs]
+    return jobs_list[:limit]
```
Comment (wardi) [inline ckan/logic/action/get.py:3564]: avoid unnecessary dictize_job calls
```suggestion
            jobs_list += [jobs.dictize_job(j) for j in queue.jobs[:limit - len(jobs_list)]
    return jobs_list
```

File: ckan/logic/schema.py [hunk 3]
```diff
@@ -790,9 +790,10 @@ def update_configuration_schema():
 
 
 @validator_args
-def job_list_schema(ignore_missing, list_of_strings):
+def job_list_schema(ignore_missing, list_of_strings, int_validator):
     return {
         u'queues': [ignore_missing, list_of_strings],
+        u'limit': [ignore_missing, int_validator],
```
Comment (wardi) [inline ckan/logic/schema.py:796]: ids_only?

**Comment by JVickery-TBS** [comment]
@wardi okay this should be good now I think

**Review by wardi — APPROVED** [review]
looks good, should be good for upstream too with fewer `u` prefixes

**Pull request merged**